From 668ac03964dd80712ef5326e17a23f46891d52e6 Mon Sep 17 00:00:00 2001
From: Jianqun Xu <jay.xu@rock-chips.com>
Date: Tue, 11 Apr 2017 10:58:53 +0800
Subject: [PATCH] arm: dts: rk3288: add mipi support

if want enable mipi,
you should to okay mipi_dsi, dsi_panel and route_mipi.
also you should to disabled edp, edp_phy and edp_panel.

Change-Id: I037cb76c7a1d08bcd09219072aeb359e0d460a1c
Signed-off-by: xubilv <xbl@rock-chips.com>
Signed-off-by: Jianqun Xu <jay.xu@rock-chips.com>
---
 arch/arm/boot/dts/rk3288-android.dtsi | 75 +++++++++++++++++++++++++++
 arch/arm/boot/dts/rk3288.dtsi         |  1 +
 2 files changed, 76 insertions(+)

diff --git a/arch/arm/boot/dts/rk3288-android.dtsi b/arch/arm/boot/dts/rk3288-android.dtsi
index fbaee75d1e10..f717cfd29391 100644
--- a/arch/arm/boot/dts/rk3288-android.dtsi
+++ b/arch/arm/boot/dts/rk3288-android.dtsi
@@ -122,6 +122,22 @@
 		pwms = <&pwm0 0 1000000 PWM_POLARITY_INVERTED>;
 	};
 
+	display-subsystem {
+		status = "okay";
+
+		ports = <&vopb_out>, <&vopl_out>;
+		route {
+			route_mipi: route-mipi {
+				status = "disabled";
+				logo,uboot = "logo.bmp";
+				logo,kernel = "logo_kernel.bmp";
+				logo,mode = "center";
+				charge_logo,mode = "center";
+				connect = <&vopb_out_mipi>;
+			};
+		};
+	};
+
 	rk_key: rockchip-key {
 		compatible = "rockchip,key";
 		status = "okay";
@@ -256,6 +272,41 @@
 	compatible = "rockchip,rk3288-secure-efuse";
 };
 
+&mipi_dsi {
+	dsi_panel: dsi-panel {
+		compatible = "simple-panel-dsi";
+		reg = <0>;
+		backlight = <&backlight>;
+		enable-gpios = <&gpio7 4 GPIO_ACTIVE_HIGH>;
+		power-supply = <&vcc_lcd>;
+		dsi,flags = <(MIPI_DSI_MODE_VIDEO | MIPI_DSI_MODE_VIDEO_BURST)>;
+		dsi,format = <MIPI_DSI_FMT_RGB888>;
+		dsi,lanes = <4>;
+		delay,prepare = <120>;
+		status = "disabled";
+
+		disp_timings: display-timings {
+			native-mode = <&timing0>;
+
+			timing0: timing0 {
+				clock-frequency = <150000000>;
+				hactive = <1200>;
+				vactive = <1920>;
+				hback-porch = <80>;
+				hfront-porch = <81>;
+				vback-porch = <21>;
+				vfront-porch = <21>;
+				hsync-len = <10>;
+				vsync-len = <3>;
+				hsync-active = <0>;
+				vsync-active = <0>;
+				de-active = <0>;
+				pixelclk-active = <0>;
+			};
+		};
+	};
+};
+
 &rga {
 	compatible = "rockchip,rga2";
 	clocks = <&cru ACLK_RGA>, <&cru HCLK_RGA>, <&cru SCLK_RGA>;
@@ -274,6 +325,30 @@
 	rockchip,usb-mode = <0>;
 };
 
+&edp {
+	status = "okay";
+};
+
+&edp_phy {
+	status = "okay";
+};
+
+&edp_panel {
+	status = "okay";
+};
+
+&dsi_panel {
+	status = "disabled";
+};
+
+&mipi_dsi {
+	status = "disabled";
+};
+
+&route_mipi {
+	status = "disabled";
+};
+
 &pinctrl {
 
 	backlight {
diff --git a/arch/arm/boot/dts/rk3288.dtsi b/arch/arm/boot/dts/rk3288.dtsi
index 79e10bcf530d..1947ef8f9ceb 100644
--- a/arch/arm/boot/dts/rk3288.dtsi
+++ b/arch/arm/boot/dts/rk3288.dtsi
@@ -46,6 +46,7 @@
 #include <dt-bindings/thermal/thermal.h>
 #include <dt-bindings/power/rk3288-power.h>
 #include <dt-bindings/soc/rockchip,boot-mode.h>
+#include <dt-bindings/display/drm_mipi_dsi.h>
 #include "skeleton.dtsi"
 
 / {
-- 
2.35.3

